crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4222,21 +4218,10 @@ where
             flake8_pie::rules::no_unnecessary_pass(self, body);
         }
 
-        // Phase 2: Binding
-        let prev_body = self.semantic.body;
```

Comment:
Nit: Probably not worth doing. We could make this approach slightly faster if we track the *branch* we're currently in. This should be less problematic because we only need to write the data when we take a new branch. Don't we already have something similar?